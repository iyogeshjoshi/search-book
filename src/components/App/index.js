import React from "react";
import "./App.css";
import Home from "../Home";

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h2>Search Book</h2>
      </header>

      <Home />
    </div>
  );
}

export default App;
