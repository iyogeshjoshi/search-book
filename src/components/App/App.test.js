import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "./";

describe("App renders properly", () => {
  test("renders learn react link", () => {
    const { getByText } = render(<App />);
    const linkElement = getByText(/search book/i);

    expect(linkElement).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
