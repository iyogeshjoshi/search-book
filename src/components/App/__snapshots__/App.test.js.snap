// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h2>
      Search Book
    </h2>
  </header>
  <div
    className="main-container"
  >
    <section
      className="search-form"
    >
      <input
        defaultValue={3}
        id="quantity"
        min={1}
        name="quantity"
        onChange={[Function]}
        title="number of results to show"
        type="number"
      />
      <input
        autoFocus={true}
        id="query"
        list="titles"
        name="query"
        onChange={[Function]}
        placeholder="search here"
        tabIndex="1"
        title="Enter your input here to search books"
        type="search"
      />
      <div
        className="dropdown false"
        id="titles"
      />
    </section>
    <section
      className="search-result"
    />
  </div>
</div>
`;
